pkg/controller/vsphere/reconciler.go

```diff
@@ -341,6 +341,32 @@ func clone(s *machineScope) error {
 		return err
 	}
 
+	var snapshotRef *types.ManagedObjectReference
+
+	// If a linked clone is requested then a MoRef for a snapshot must be
+	// found with which to perform the linked clone.
+	// Empty clone mode is linked clone
+	if s.providerSpec.CloneMode == "" || s.providerSpec.CloneMode == vsphereapi.LinkedClone {
+		if s.providerSpec.Snapshot == "" {
+			klog.V(3).Infof("%v: no snapshot name provided, getting snapshot using template", s.machine.GetName())
+			var vm mo.VirtualMachine
+			if err := vmTemplate.Properties(s.Context, vmTemplate.Reference(), []string{"snapshot"}, &vm); err != nil {
+				return errors.Wrapf(err, "error getting snapshot information for template %s", vmTemplate.Name())
+			}
+
+			if vm.Snapshot != nil {
+				snapshotRef = vm.Snapshot.CurrentSnapshot
+			}
+		} else {
+			klog.V(3).Infof("%v: searching for snapshot by name %s", s.machine.GetName(), s.providerSpec.Snapshot)
+			var err error
+			snapshotRef, err = vmTemplate.FindSnapshot(s.Context, s.providerSpec.Snapshot)
+			if err != nil {
+				klog.V(3).Infof("%v: failed to find snapshot %s", s.machine.GetName(), s.providerSpec.Snapshot)
+			}
+		}
+	}
+
```

Comment:
shouldn't we set the `diskMoveType` based on the above
e.g
```
fullCloneDiskMoveType = types.VirtualMachineRelocateDiskMoveOptionsMoveAllDiskBackingsAndConsolidate
	linkCloneDiskMoveType = types.VirtualMachineRelocateDiskMoveOptionsCreateNewChildDiskBacking
```
See https://github.com/openshift/machine-api-operator/pull/512/files#r394223024

Reply:
👍